pkgs/development/python-modules/pip/default.nix

```diff
@@ -36,18 +43,50 @@ buildPythonPackage rec {
     installShellFiles
     setuptools
     wheel
+
+    # docs
+    sphinx
+    sphinx-copybutton
+    sphinx-inline-tabs
+    sphinxcontrib-towncrier
+    myst-parser
   ];
 
+  outputs = [
+    "out"
+    "man"
+  ];
+
+  preBuild = ''
+    cd docs
+    # help custom docs code find pip source
+    PYTHONPATH=$src/src:$PYTHONPATH
+
+    python pip_sphinxext.py
+
+    # explicitly setting intersphinx_mapping. to invalid files to avoid network timeouts
+    # in theory the right files could come from the nix store
+    sphinx-build -b man \
```

Comment:
Have you using `sphinxHook` instead? Just out of curiosity.

Reply:
Initially I looked at it as incompatible because of the pre-req coming from `python pip_sphinxext.py`. I'll have a look at using the hook just for this step and let you know.